public/app/features/correlations/Forms/types.ts

```diff
@@ -1,11 +1,12 @@
-import { Correlation } from '../types';
+import { CorrelationConfig } from '../types';
 
 export interface FormDTO {
   sourceUID: string;
   targetUID: string;
   label: string;
   description: string;
+  config: CorrelationConfig;
 }
 
-type FormDTOWithoutTarget = Omit<FormDTO, 'targetUID'>;
-export type EditFormDTO = Partial<FormDTOWithoutTarget> & Pick<FormDTO, 'sourceUID'> & { uid: Correlation['uid'] };
+type FormDTOWithoutTarget = Omit<FormDTO, 'targetUID' | 'sourceUID'>;
+export type EditFormDTO = FormDTOWithoutTarget;
```

Comment:
It's much simpler so probably can be inlined now.

```suggestion
export type EditFormDTO = Omit<FormDTO, 'targetUID' | 'sourceUID'>;
```